### ISSUE TITLE
Config changes may fail if root is enabled

### ISSUE THREAD

**Issue by RunningDroid** [syncthing/syncthing-android#434]
If I change the device name and select reboot now something breaks and makes the app unusable, the only changes I made to the default config were to add a device and enable sync as root.

[Logcat](https://gist.github.com/RunningDroid/d09ad64bd153e9218eec)

**Comment by Nutomic** [comment]
The warning comes from the Syncthing key generation. We seem to delete the config because it is invalid, and Syncthing generates a new one.

I guess the config could be owned by root in your case.  Can you check the permissions of the config file in `/data/data/com.nutomic.syncthingandroid/files`?

**Comment by RunningDroid** [comment]
`/data/data/com.nutomic.syncthingandroid/files/config.xml` and `config.xml.invalid` are 0600 and owned by root, everything else is 0600 and owned by uid 10113.

I tried reproducing this again and changing the device name without root works, but changing the device name after I enable root hits the bug.

**Comment by Nutomic** [comment]
Ah, I forgot to actually enable root. 

Looks like changing a setting causes Syncthing to rewrite the config, so that it is owned by root. But we need to read the config at start (mainly to get the address and api key). So we probably have to chmod the config file to make it publicly readable.

**Comment by DaveQB** [comment]
I had the same or similar issue.
I found the config.xml was still owned by the right user (not root) and 600 perms.

I tried going in an editing the "com.nutomic.syncthingandroid_preferences.xml" file and setting the use root value to false from true, but no difference (even rebooted the phone to remove syncthing from any active memory locations).

So I backed up all in /data/data/com.nutomic.syncthingandroid/, removed the app, re-installed, restored all the keys and certs. I then tried to copy in my config.xml but this rendered the app unable to start. So I had to config it all through the apps gui.

I won't do that again! (setting to use root perms)
